src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -121,40 +161,138 @@ public void Initialize(StorageEnvironment environment, TransactionContextPool co
             }
         }
 
-        public LicenseStatus GetLicenseStatus()
+        public async Task ReloadLicenseLimits()
         {
-            var lastLicenseIndex = _serverStore.LastLicenseIndex;
-            if (lastLicenseIndex > _lastSavedLicenseIndex)
+            try
             {
-                _lastSavedLicenseIndex = lastLicenseIndex;
-                ReloadLicense();
+                var licenseLimits = _serverStore.LoadLicenseLimits();
+                if (licenseLimits?.CoresByNode != null && 
+                    licenseLimits.CoresByNode.TryGetValue(_serverStore.NodeTag, out var cores))
+                {
+                    cores = Math.Min(cores, _licenseStatus.MaxCores);
+
+                    var process = Process.GetCurrentProcess();
+                    SetAffinity(process, cores);
+
+                    var ratio = (int)Math.Ceiling(_licenseStatus.MaxRamInGb / (double)_licenseStatus.MaxCores);
+                    var clusterSize = GetClusterSize();
+                    var maxWorkingSet = Math.Min(_licenseStatus.MaxRamInGb / (double)clusterSize, cores * ratio);
```

Comment:
Why this line? If I have a 6 core cluster licenses, and I have 3 nodes, I want to be able to have a node with 4 cores and 2 with 1 core each.
This doesn't seem to allow it to follow the same RAM allocation.